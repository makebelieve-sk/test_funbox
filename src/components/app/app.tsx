import React, { useState } from 'react';

import { mockData } from '../../mock-data/mock-data';
import { CardComponent } from '../card-component/card-component';

export const App: React.FC = () => {
  const [selected, setSelected] = useState<boolean[]>(new Array(mockData.length).fill(false));
  const [buyMode, setBuyMode] = useState<null | string>(null);
  let text;

  const mouseLeave = (i: number, setTitle: React.Dispatch<React.SetStateAction<string | null>>) => {
    if (selected.find((status: boolean, j: number) => j === i)) {
        setTitle(mockData[i].hoverTitle);
    }
  };

  const mouseEnter = (i: number, setTitle: React.Dispatch<React.SetStateAction<string | null>>) => {
      if (selected.find((status: boolean, j: number) => j === i)) {
          setTitle(mockData[i].title);
      }
  };

  const change = (i: number) => {
    const selectedarr = [...selected];
    selectedarr[i] = !selected[i];
    setSelected(selectedarr);
  };

  return (
    <div className="main_wrapper">
      <div className="shadow">
        <h1 className="header">Ты сегодня покормил кота?</h1>
        <div className="block_products">
          {mockData.map((product, i) => {
            if (product.buy) {
              text = (
                <>
                  { product.text }
                  <a href="/#" className="buy_link" onClick={(e) => {
                    e.preventDefault();
                    const selectedarr = [...selected];
                    selectedarr[i] = !selected[i];
                    setSelected(selectedarr);
                    setBuyMode('choosen');
                  }}>купи.</a>
                </>
              );
            } else {
              text = product.text;
            }

            if (product.quantity > 0) {
              return (
                <CardComponent
                  key={`${product.id}`}
                  product={product}
                  selected={selected}
                  checked={true}
                  buyMode={buyMode} 
                  i={i} 
                  text={text}
                  mouseLeave={mouseLeave}
                  mouseEnter={mouseEnter}
                  change={change}
                  disabled={null}
                />
              );
            } else {
              text = `Печалька, ${product.with} закончился.`;

              return (
                <CardComponent
                  key={`${product.id}`}
                  product={product}
                  selected={selected}
                  checked={null}
                  buyMode={buyMode} 
                  i={i} 
                  text={text}
                  mouseLeave={null}
                  mouseEnter={null}
                  change={null}
                  disabled={'disabled'}
                />
              );
            }          
          })}        
        </div>
      </div>
    </div>
  );
};