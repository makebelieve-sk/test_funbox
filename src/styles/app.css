.root {
  height: 100%;
  width: 100%;
}
.header {
  font-size: 48px;
  font-family: 'Exo 2.0 Thin', arial;;
  color: rgb(255, 255, 255);
  line-height: 1.2;
  text-align: center;
  text-shadow: 0px 1px 0px rgba(0, 0, 0, 0.004);
  z-index: 174;
  margin: 0;
  padding-top: 36px;
}

.block_products {
  display: flex;
  justify-content: space-around;
}
.choose_input {
  opacity: 0;
} 
.choose_check {
  top: 104px;
  width: 312px;
  height: 472px;
  z-index: 51;
  position: relative;
  height: 480px;
  width: 320px;
  display: block;
}
input[type="checkbox"] + span {
  position: absolute;
  left: 0; top: 0;
  width: 100%; height: 100%;
  background: url('./img/1.png') no-repeat; 
  cursor: pointer;
}
input[type="checkbox"]:checked + span {
  background: url('./img/2.png') no-repeat;
}
.choosen {
  background: url('./img/2.png') no-repeat;
}
input[type="checkbox"].disabled + span {
  width: 100%; height: 100%;
  background: url('./img/3.png') no-repeat; 
  cursor: pointer;
}
.product_weight {
  position: absolute;
  right: 26px;
  bottom: 27px;   
  height: 53px;
  width: 60px;
  line-height: 10.524px;
  cursor: pointer;
}
.product_weight_number {
  margin: 0;
  margin-top: 10.524px;
  font-size: 42px;
  font-family: "Trebuchet MS";
  color: rgb(255, 255, 255);
  text-align: center;
  z-index: 54; 
}
.product_weight_text {  
  font-size: 21px;
  font-family: "Trebuchet MS";
  color: rgb(255, 255, 255);
  text-align: center;
  z-index: 54;
}
.buy_link {
  color: #1698d9;
  text-decoration: none;
  border-bottom: 1px dashed #1698d9;
}
.block_in_card {
  position: absolute;
  left: 40px;
  top: 0;
  width: 75%;
  height: 50%;
  margin: 0 auto;
  cursor: pointer;
}
.product_title {
  font-size: 16px;
  font-family: "Trebuchet MS";
  color: rgb(102, 102, 102);
  line-height: 1.2;
  text-align: left;
  z-index: 58;
  width: 100%;
  margin: 0;
  margin-top: 24px;  
}
.product_name {
  font-size: 48px;
  font-family: "Trebuchet MS";
  color: rgb(0, 0, 0);
  font-weight: bold;
  line-height: 1.2;
  text-align: left;
  z-index: 57;
  margin: 0;
}
.product_with {
  font-size: 24px;
  font-family: "Trebuchet MS";
  color: rgb(0, 0, 0);
  font-weight: bold;
  line-height: 1.2;
  text-align: left;
  z-index: 56;
  margin: 0;
  margin-bottom: 14px;
}
.product_description {
  font-size: 14px;
  font-family: "Trebuchet MS";
  color: rgb(102, 102, 102);
  font-weight: bold;
  line-height: 1.143;
  text-align: left;
  z-index: 55;
  margin: 0;
}
.product_text {
  font-size: 13px;
  font-family: "Trebuchet MS";
  color: rgb(255, 255, 255);
  line-height: 1.231;
  text-align: center;
  text-shadow: 0px 1px 0px rgba(0, 0, 0, 0.004);
  position: absolute;
  top: 506.349px;
  z-index: 49;
  width: 100%;
  margin: 0;
}
.disabled_text {
  font-size: 13px;
  font-family: "Trebuchet MS";
  color: rgb(255, 255, 102);
  line-height: 1.231;
  text-align: center;
  text-shadow: 0px 1px 0px rgba(0, 0, 0, 0.004);
  z-index: 118;
  position: absolute;
  top: 506.349px;
  z-index: 49;
  width: 100%;
  margin: 0;
}
.disabled {
  color: rgba(179, 179, 179, 0.502);
}
body {
  height: 100%;
  background-image: url("./img/Pattern.png");
  z-index: 1;
}
.shadow {
  background-image: url("./img/Shadow.png") no-repeat;
  width: 100vw;
  height: 100vh;
  background-size: 100%;
  z-index: 2;
  opacity: 1;
}
@media (max-width: 1200px) {
  .block_products {
    display: block;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }
  .product_wrapper {
    display: block;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }
  label {
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 80px;
  }
 }