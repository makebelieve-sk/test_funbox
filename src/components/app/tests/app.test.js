import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { App } from '../app';

Enzyme.configure({ adapter: new Adapter() });

const setUp = () => shallow(<App />);

describe(`App component`, () => {
    let component;

    beforeEach(() => {
        component = setUp();
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.main_wrapper');
        expect(wrapper.length).toBe(1);
    });

    it(`App correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<App />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});