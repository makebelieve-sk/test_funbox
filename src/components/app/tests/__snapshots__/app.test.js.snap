// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App correctly renders after relaunch 1`] = `
<div
  className="main_wrapper"
>
  <div
    className="shadow"
  >
    <h1
      className="header"
    >
      Ты сегодня покормил кота?
    </h1>
    <div
      className="block_products"
    >
      <div
        className="product-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <label
          className="choose_check"
          htmlFor="choose-0"
        >
          <input
            checked={false}
            className="choose_input null null"
            id="choose-0"
            name="choose"
            onChange={[Function]}
            type="checkbox"
          />
          <span />
          <div
            className="block_in_card"
          >
            <p
              className="product_title null"
            >
              Сказочное заморское яство
            </p>
            <p
              className="product_name null"
            >
              Нямушка
            </p>
            <p
              className="product_with null"
            >
              с фуа-гра
            </p>
            <p
              className="product_description null"
            >
              10 порций
            </p>
            <p
              className="product_description null"
            >
              мышь в подарок
            </p>
            <p
              className="product_description null"
            />
          </div>
          <div
            className="product_weight"
          >
            <p
              className="product_weight_number"
            >
              0,5
            </p>
            <p
              className="product_weight_text"
            >
              кг
            </p>
          </div>
          <p
            className="product_text null_text"
          >
            Чего сидишь? Порадуй котэ, 
            <a
              className="buy_link"
              href="/#"
              onClick={[Function]}
            >
              купи.
            </a>
          </p>
        </label>
      </div>
      <div
        className="product-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <label
          className="choose_check"
          htmlFor="choose-1"
        >
          <input
            checked={false}
            className="choose_input null null"
            id="choose-1"
            name="choose"
            onChange={[Function]}
            type="checkbox"
          />
          <span />
          <div
            className="block_in_card"
          >
            <p
              className="product_title null"
            >
              Сказочное заморское яство
            </p>
            <p
              className="product_name null"
            >
              Нямушка
            </p>
            <p
              className="product_with null"
            >
              с рыбой
            </p>
            <p
              className="product_description null"
            >
              40 порций
            </p>
            <p
              className="product_description null"
            >
              2 мыши в подарок
            </p>
            <p
              className="product_description null"
            />
          </div>
          <div
            className="product_weight"
          >
            <p
              className="product_weight_number"
            >
              2
            </p>
            <p
              className="product_weight_text"
            >
              кг
            </p>
          </div>
          <p
            className="product_text null_text"
          >
            Головы щучьи с чесноком да свежайшая сёмгушка.
          </p>
        </label>
      </div>
      <div
        className="product-wrapper"
        onMouseEnter={null}
        onMouseLeave={null}
      >
        <label
          className="choose_check"
          htmlFor="choose-2"
        >
          <input
            checked={null}
            className="choose_input null disabled"
            id="choose-2"
            name="choose"
            onChange={null}
            type="checkbox"
          />
          <span />
          <div
            className="block_in_card"
          >
            <p
              className="product_title disabled"
            >
              Сказочное заморское яство
            </p>
            <p
              className="product_name disabled"
            >
              Нямушка
            </p>
            <p
              className="product_with disabled"
            >
              с курой
            </p>
            <p
              className="product_description disabled"
            >
              100 порций
            </p>
            <p
              className="product_description disabled"
            >
              5 мышей в подарок
            </p>
            <p
              className="product_description disabled"
            >
              заказчик доволен
            </p>
          </div>
          <div
            className="product_weight"
          >
            <p
              className="product_weight_number"
            >
              5
            </p>
            <p
              className="product_weight_text"
            >
              кг
            </p>
          </div>
          <p
            className="product_text disabled_text"
          >
            Печалька, с курой закончился.
          </p>
        </label>
      </div>
    </div>
  </div>
</div>
`;
